components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/certificatechecktrigger/certificatechecktrigger_test.go

```diff
@@ -0,0 +1,301 @@
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"14.15.16.17",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				}
+			]
+		  }
+		}`), &ipConfig)
+	return ipConfig
+}
+
+func GetCertificateRequest() models.CertificateCheckRequest {
+	req := models.CertificateCheckRequest{}
+	json.Unmarshal([]byte(`{
+		"RootCertificate": "---- VALID ROOT CA ----",
+		"PrivateKey": "---- VALID PRIVATE KEY ----",
+		"NodeCertificate": "---- VALID NODE CERT ----",
+		"AdminPrivateKey": "---- VALID ADMIN PRIVATE KEY ----",
+		"AdminCertificate": "---- VALID ADMIN CERT ----"
+	 }`), &req)
+	return req
+}
+
+func TestNewCertificateCheck(t *testing.T) {
+	testPort := "1234"
+	cc := NewCertificateCheck(logger.NewTestLogger(), testPort)
+	assert.NotNil(t, cc)
+	assert.NotNil(t, cc.log)
+	assert.Equal(t, constants.LOCAL_HOST_URL, cc.host)
+	assert.Equal(t, testPort, cc.port)
+}
+
+func startMockServerOnCustomPort(mockServer *httptest.Server, port string) error {
+	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%s", port))
+	if err != nil {
+		return err
+	}
+	mockServer.Listener = l
+	mockServer.Start()
+	return nil
+}
+
+func TestCertificateCheck_Run(t *testing.T) {
+
+	t.Run("Returns OK", func(t *testing.T) {
+		//starting the mock server on custom port
+		mockServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte(certificateCheckSuccessResp))
+		}))
+		err := startMockServerOnCustomPort(mockServer, "1234")
+		assert.NoError(t, err)
+		defer mockServer.Close()
+
+		cc := NewCertificateCheck(logger.NewTestLogger(), "1234")
+		request := GetRequestJson()
+		finalResp := cc.Run(request)
+		totalIps := request.Hardware.AutomateNodeCount + request.Hardware.ChefInfraServerNodeCount + request.Hardware.PostgresqlNodeCount + request.Hardware.OpenSearchNodeCount
+		assert.Equal(t, totalIps, len(finalResp))
+
+		for _, resp := range finalResp {
+			assert.Equal(t, constants.CERTIFICATE, resp.Result.Check)
+			assert.Equal(t, constants.CERTIFICATE_MSG, resp.Result.Message)
+			assert.NotEmpty(t, resp.Result.Checks)
+			if resp.Host == "14.15.16.17" {
+				triggerResp := resp
+				assert.Equal(t, "SUCCESS", triggerResp.Status)
+				assert.NotEmpty(t, triggerResp.Result)
+				assert.Equal(t, resp.Result.Passed, true)
+				assert.Equal(t, 4, len(triggerResp.Result.Checks))
+				assert.Equal(t, "Certificates have valid expiry date", triggerResp.Result.Checks[0].Title)
+				assert.Equal(t, true, triggerResp.Result.Checks[0].Passed)
+				assert.Equal(t, "All certificates expiry date is later than 365 days", triggerResp.Result.Checks[0].SuccessMsg)
+			} else {
+				assert.Equal(t, resp.Result.Passed, true)
+			}
+		}
+	})
+
+	t.Run("Failure response", func(t *testing.T) {
```

Comment:
We need to add failure response for Status code other than 200

Reply:
Added for 500 internal server error